### PR TITLE
Add Callout component. Update Button component.

### LINKED ISSUE #42: Update Component: Button

### LINKED ISSUE #88: New Component: Callout

### LINKED ISSUE #98: CompactGrouper prop radius warning
React is throwing a warning on the CompactGrouper prop radius.

```
Warning: Invalid value for prop `radius` on <div> tag. Either remove it from the element, or pass a string or number value to keep it in the DOM
```

per https://github.com/raster-foundry/blasterjs/pull/97#issuecomment-438391450

### PR DESCRIPTION
**Pull request by lederer**
## Overview

Introduce the `Callout` component. Update the `Button` component to make use of `Icon` component and latest enums and to adhere to our latest code practices.

Both components accept an `intent` prop that expects the `Intent` constant and determines the component's color and, in `Callout`'s case, icon. `Callout`'s icon can be overridden via the `icon` prop, which expects a string matching the name of a known icon (or `false` to cancel the `intent`'s default icon).

`Button` accepts up to two icons ( `iconBefore` and `iconAfter`), along with an `appearance` prop that expects the `Appearance` constant (currently `DEFAULT`, `MINIMAL`, `PROMINENT`) and determines foreground and background colors, factoring in the `intent`'s color.

See each component's docs page for further information on props and states.

### Checklist

- [x] Relevant documentation pages have been created or updated
- [ ] Description of PR is in an appropriate section of the changelog and grouped with similar changes if possible

### Demo

image

image

image

image

image

image

image

image

image

### Notes

I also did a drive-by bugfix on the Grouper component (#98).

## Testing Instructions

 * Evaluate the callout and button docs pages
 * Pay particular attention to button states (hover, focus, active, disabled)

Closes #88 
Closes #42 
Closed #98

### CODE REVIEW

**Comment by designmatty** [comment]
ohhhhh. this is dope looking so far. I'm gonna try and get this merged in today.

**Comment by alkamin** [comment]
:tada: My only nitpick is the callout icon alignment when there is only a single line of text:

image

**Comment by lederer** [comment]
> My only nitpick is the callout icon alignment when there is only a single line of text

Yeah. Though it seems to depend on the icon. Eg, these are fine as is …

image

image

If I nudge them down, they look wrong, though the `code` icon looks right. 

Is there something we can do to normalize the size and position of the icons within their viewBox?

**Comment by designmatty** [comment]
There is definitely optimizations that need to happen to icons. I'm going to say that we won't be able to have viewport be the same across the board but we can definitely do better with optical sizing issues.

File: packages/core/components/callout/index.mdx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        With intent (DANGER) and custom icon, overriding intent's default icon.
+    </Callout>
+</Playground>
+
+
+## Icon Size
+
+<Playground>
+    <Callout
+        title="I am a banana"
```
Comment (designmatty) [inline packages/core/components/callout/index.mdx:62]: LOLOL wat?

File: packages/core/common/appearance.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+export const Appearance = {
+  DEFAULT: "default",
+  PROMINENT: "prominent",
```
Comment (designmatty) [inline packages/core/common/appearance.js:3]: I like this choice

**Review by designmatty — APPROVED** [review]
Looks good!

File: packages/core/components/callout/index.mdx (hunk 1 above)
Comment (alkamin) [reply packages/core/components/callout/index.mdx:62]: I did actually LOL the first time I saw this.

**Pull request merged**